lib/diff-array.js

```diff
@@ -191,6 +200,9 @@ var handleDeepProperties = function(object, paths, cb) {
     var currentObject = object;
 
     for (var i = 0; i < props.length - 1; i++) {
+      if(_.isUndefined(currentObject[props[i]])){
```

Comment:
OK, I've just pushed a few fixes to a new branch (was suppose to be unrelated to your changes) here: https://github.com/Urigo/angular-meteor/commit/9ad36f8784abd7669add6918aa5deec52dfbe37b

Can you have a short look to make sure we are ok and it's not related?
thanks